psl-core/src/main/java/org/linqs/psl/util/VizDataCollection.java

```diff
@@ -0,0 +1,158 @@
+package org.linqs.psl.util;
+
+import org.linqs.psl.grounding.GroundRuleStore;
+import org.linqs.psl.model.atom.Atom;
+import org.linqs.psl.model.atom.GroundAtom;
+import org.linqs.psl.model.rule.GroundRule;
+import org.linqs.psl.model.rule.WeightedGroundRule;
+import org.linqs.psl.model.rule.logical.AbstractLogicalRule;
+import org.linqs.psl.model.term.Constant;
+import org.linqs.psl.model.term.Variable;
+
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+
+
+public class VizDataCollection {
+    private static Runtime runtime = null;
+    private static VisualizationData vizData = null;
+
+    private static final Logger log = LoggerFactory.getLogger(VizDataCollection.class);
+
+    public static String outputPath = null;
+
+    static {
+        init();
+    }
+
+    private VizDataCollection() {}
+
+    private static synchronized void init() {
+        if (runtime != null) {
+            return;
+        }
+        vizData = new VisualizationData();
+        runtime = Runtime.getRuntime();
+        runtime.addShutdownHook(new ShutdownHook());
+    }
+
+    public static void outputJSON() {
+        String[] keyNames = {"truthMap", "rules", "groundRules", "groundAtoms"};
+        JSONObject fullJson = new JSONObject(vizData, keyNames);
+
+        PrintStream stream = System.out;
+        boolean closeStream = false;
+
+        if (outputPath != null) {
+            try {
+                stream = new PrintStream(outputPath);
+                closeStream = true;
+            } catch (IOException ex) {
+                log.error(String.format("Unable to open file (%s) for visualization data," +
```

Comment:
Instead of falling back, just fail. Rethrow this exception wrapped in a RuntimeException.